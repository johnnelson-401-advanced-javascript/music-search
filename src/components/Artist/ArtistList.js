import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

export default function ArtistList({ artists, handlePageBackward, handlePageForward }) {

  const artistElements = artists.map(artist => (
    <li key={artist.id}>
      <Artist name={artist.name} id={artist.id} disamb={artist.disamb} />
    </li>
  ));

  return (
    <>
      <button onClick={handlePageBackward}>Back</button>
      <button onClick={handlePageForward}>Next</button>
      <ul>
        {artistElements}
      </ul>
    </>
  );
}
ArtistList.propTypes = {
  handlePageBackward: PropTypes.func.isRequired,
  handlePageForward: PropTypes.func.isRequired,
  artists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    disamb: PropTypes.string,
  })).isRequired
};
