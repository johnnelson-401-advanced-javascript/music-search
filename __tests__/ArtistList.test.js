import React from 'react';
import { shallow } from 'enzyme';
import ArtistList from '../src/components/Artist/ArtistList';

describe('ArtistList component', () => {
  it('renders ArtistList', () => {
    const wrapper = shallow(<ArtistList 
      handlePageForward={()=>{}} 
      handlePageBackward={()=>{}} 
      artists={[{ id: '1', name: 'Luke' }]} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
