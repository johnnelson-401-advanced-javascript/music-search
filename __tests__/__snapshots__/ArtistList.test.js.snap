// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component renders ArtistList 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    Back
  </button>
  <button
    onClick={[Function]}
  >
    Next
  </button>
  <ul>
    <li
      key="1"
    >
      <Artist
        id="1"
        name="Luke"
      />
    </li>
  </ul>
</Fragment>
`;
